jraph_MPEU/train.py

```diff
@@ -407,6 +408,45 @@ def create_optimizer(
     raise ValueError(f'Unsupported optimizer: {config.optimizer}.')
 
 
+def loss_fn_mse(params, state, rng, graphs, net_apply):
+    """Mean squared error loss function for regression."""
+    labels = graphs.globals
+    graphs = replace_globals(graphs)
+
+    mask = get_valid_mask(graphs)
+    pred_graphs, new_state = net_apply(params, state, rng, graphs)
+    predictions = pred_graphs.globals
+    labels = jnp.expand_dims(labels, 1)
+    sq_diff = jnp.square((predictions - labels)*mask)
+
+    loss = jnp.sum(sq_diff)
+    mean_loss = loss / jnp.sum(mask)
+    absolute_error = jnp.sum(jnp.abs((predictions - labels)*mask))
+    mae = absolute_error /jnp.sum(mask)
+
+    return mean_loss, (mae, new_state)
+
+
+def loss_fn_bce(params, state, rng, graphs, net_apply):
+    """Binary cross entropy loss function for classification."""
+    labels = graphs.globals
+    graphs = replace_globals(graphs)
+    targets = jax.nn.one_hot(labels, 2)
+
+    # try get_valid_mask function instead
+    mask = jraph.get_graph_padding_mask(graphs)
+    pred_graphs, new_state = net_apply(params, state, rng, graphs)
```

Comment:
It looks like from your test the new_state=state when we call net_apply. This seems to match with haiku docs: where they say the most common usage of state is to `represent the moving averages used in batch normalization`. So for us this variable is not used since we're not doing batch norm right?